components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/services/UserIdentityManagementAdminService.java

```diff
@@ -83,6 +85,11 @@ public void deleteUser(String userName) throws IdentityMgtServiceException {
         try {
             UserStoreManager userStoreManager = IdentityMgtServiceComponent.getRealmService().
                     getTenantUserRealm(CarbonContext.getThreadLocalCarbonContext().getTenantId()).getUserStoreManager();
+            if (isAttemptToUpdateSuperAdmin(userName)) {
+                log.warn("An attempt to delete admin user.");
+                throw new org.wso2.carbon.user.core.UserStoreException("You do not have the required privilege to " +
```

Comment:
`org.wso2.carbon.user.core.UserStoreException` seems to be not declared as thrown from the method.

Reply:
This is since we catch the 'UserStoreException' within the method and throw an `IdentityMgtServiceException`.

Reply:
Either we can throw the same `IdentityMgtServiceException` or `UserStoreException` which is already imported in the class.